File: my-folder/0121-best-time-to-buy-and-sell-stock/solution.ts
function maxProfit(prices: number[]): number {
  let maxProfit = 0;
  let minPrice = Number.MAX_VALUE;
  for (const price of prices) {
    if (price < minPrice) {
      minPrice = price;
    }
    maxProfit = Math.max(maxProfit, price - minPrice);
  }
  return maxProfit;
}

File: my-folder/0121-best-time-to-buy-and-sell-stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxProfit: int = 0
        minPrice: int = prices[0]
        for price in prices:
            if price < minPrice:
                minPrice = price
            maxProfit = max(maxProfit, price - minPrice)
        return maxProfit


